### DISCUSSION TITLE
[Request] Trilium Notes

### DISCUSSION THREAD

**Discussion by l3gitpanda** [tteck/Proxmox#408] [Ideas]
Awesome self-hosted note taking software. Very modular so can be used for many things.

Github Page:
https://github.com/zadam/trilium
Server Install:
https://github.com/zadam/trilium/wiki/Docker-server-installation
Video showing it off:
https://www.youtube.com/watch?v=2orhoSiy-zk&ab_channel=DBTech

**Comment by timdonovanuk** [comment]
+1. I love this software!

**Comment by tteck** [comment]
Done!

**Reply by l3gitpanda** [reply]
Thanks so much for your hard work on this project!!

**Comment by Crystal1462** [comment]
Maybe I am missing something with the update aspect of this, I am also new to proxmox.  I ran the update in the LCX console, it seemed to have worked as expected however in the app it has not updated based on version numbers.  I did shutdown and restarted the LCX just to see if that helped but did not change anything.  
Can anyone shed some light as to what I am missing here?  
Thank you

Using Proxmox 7.2-3

**Reply by tteck** [reply]
Try clearing browser cache

**Reply by tteck** [reply]
Also, Proxmox is currently at version 7.2-11

**Comment by Crystal1462** [comment]
Ive cleared cache, changed browsers even just to see if that helped or was my issue.  

I am new and not debating but my proxmox claims it is current.  Maybe that is a sign of larger issues at hand..

**Reply by tteck** [reply]
Run the Proxmox VE 7 Post Install script to fix that
But for the Trilium Notes update, the update script downloads and extracts the new files in `/opt/trilium` you should be `0.55.1` I'll look into it

**Comment by Crystal1462** [comment]
I just checked the files in the LCX against what github has for 0.55.1 and they do not match,

**Reply by tteck** [reply]
Try to update now 😉

**Reply by tteck** [reply]
Hope you're backing up first

**Reply by Crystal1462** [reply]
Ahh it was unchecked, it must of been the other repo added that allowed for it upgrade to 11.

Yes, had a backup and a snapshot just in case.

**Comment by Crystal1462** [comment]
I did the post install script, only thing I had not did myself was using the beta/test repo.  

I then did the Trilium update and it worked this time.  
Thank you for your help.

**Reply by tteck** [reply]
Don't use beta/test repo

**Comment by Pims8383** [comment]
Hello,

Trilium is in maintenance mode a new project TriliumNExt available here : 
https://github.com/TriliumNext/Notes

Is it possible to update the LXC ? 

Thank you.

**Reply by tteck** [reply]
Done https://github.com/tteck/Proxmox/commit/9c88dcdbd64a4f302acb38f14aaff8efe8467bd0 https://github.com/tteck/Proxmox/commit/9c88dcdbd64a4f302acb38f14aaff8efe8467bd0
Should be able to update to the new version.

**Reply by Pims8383** [reply]
Thanks a lot for the update

But can you confirm the trilium update to new version have to be done manually the update script do nat handle it, correct ? 

Here for other my command 

`cd /
wget -q https://github.com/TriliumNext/Notes/releases/download/v0.90.3/trilium-linux-x64-server-0.90.3.tar.xz
systemctl stop trilium.service
rm -R  /opt/trilium
mv trilium-linux-x64-server /opt/trilium
systemctl start trilium.service`

**Reply by tteck** [reply]
The update script _should_ handle the switch.

**Reply by Pims8383** [reply]
Work well thanks

**Reply by johndoe0815** [reply]
> The update script _should_ handle the switch.

Does not work for me.
I am currently on the last official version, 0.63.7.
Running you script, it announces success:
![grafik](https://github.com/user-attachments/assets/dcd35908-5f2b-4e8e-818a-62deb8ac9d49)

But Trilium is not accessible anymore afterwards, I just get a connection error when opening IP:8080.
I repeated the process from a previous backup, rebooted, no change, after update Trilium is down.
Any idea?

**Reply by oOStroudyOo** [reply]
I can confirm that updating from Trilium to TriliumNext does not work and I'm not sure why.

Quickest way I've managed to do the upgrade so far:
```
systemctl stop trilium
rm -R /opt/trilium/
mkdir /opt/trilium/
bash -c "$(wget -qLO - https://github.com/tteck/Proxmox/raw/main/ct/trilium.sh)"
```

So seems to work with an empty directory of `/opt/trilium/` so perhaps something from the existing Trilium that doesn't get overridden when TriliumNext installs that needs to be overriden, or a permissions issue?

Upgrading from TriliumNext v0.90.3 to v0.90.4 via the Tteck scripts does work.

**Reply by johndoe0815** [reply]
Hm, when looking at other places, generally Trilium Next upgrade does not seem to cause issues when being on the lastest Trilium version 0.63.7, e.g. https://github.com/orgs/TriliumNext/discussions/369
I have not digged through the script of tteck, maybe there is anything special on top of just a normal docker install? Also curious about the checks in the script that show green checkmarks even for successful start of trilium after upgrade, while it is actually not accessible then.

**Reply by dflahaut** [reply]
I updated yesterday with the same positive message "updated successfully" ... but now when I try connect to the web http://x.x.x.x:8080, I've this error  err_connection_refused
What should I do ? I've a lot of things on the trilium server ... (-[

I tried to install a brand new lxc ... but I've the same error : err_connection_refused

**Reply by tteck** [reply]
No issues here

```
  ______     _ ___               
 /_  __/____(_) (_)_  ______ ___ 
  / / / ___/ / / / / / / __ `__ \
 / / / /  / / / / /_/ / / / / / /
/_/ /_/  /_/_/_/\__,_/_/ /_/ /_/ 
                                                                      
Using Advanced Settings
Using Distribution: debian
Using debian Version: 12
Using Container Type: 1
Using Root Password: Automatic Login
Container ID: 111
Using Hostname: trilium
Using Disk Size: 2
Allocated Cores: 1
Allocated RAM: 512
Using Bridge: vmbr0
Using IP Address: dhcp
Using Gateway IP Address: Default
Using APT-Cacher IP Address: Default
Disable IPv6: yes
Using Interface MTU Size: Default
Using DNS Search Domain: Host
Using DNS Server IP Address: Host
Using MAC Address: AE:1A:60:07:B7:59
Using Vlan: Default
Enable Root SSH Access: no
Enable Verbose Mode: no
Creating a Trilium LXC using the above advanced settings
 ✓ Using local for Template Storage.
 ✓ Using thin-pool for Container Storage.
 ✓ Updated LXC Template List
 ✓ LXC Container 111 was successfully created.
 ✓ Started LXC Container
 ✓ Set up Container OS
 ✓ Network Connected: 192.168.86.250 
 ✓ IPv4 Internet Connected
 ✗ IPv6 Internet Not Connected
 ✓ DNS Resolved github.com to 140.82.112.3
 ✓ Updated Container OS
 ✓ Installed Dependencies
 ✓ Installed TriliumNext
 ✓ Created Service
 ✓ Customized Container
 ✓ Cleaned
 ✓ Completed Successfully!

Trilium should be reachable by going to the following URL.
         http://192.168.86.250:8080 
```

![Screenshot 2024-09-14 4 27 21 AM](https://github.com/user-attachments/assets/77b92646-9321-4ce9-8965-64d0c1797e2c)

**Reply by dflahaut** [reply]
I tried succesfully a new install ... I tried to update the first/old server : no answer !! I can access with ssh on the first/old server. What should I do ?

**Reply by johndoe0815** [reply]
@tteck New install is fine, but migration from normal trilium does not work.

**Reply by dflahaut** [reply]
Any idea to restore the former install ? 
I tried to copy /root/trilium-data/document.db from former install to /root/trilium-date/document.db : KO !
I tried to copy /root/trilium-data/backup/backup-daily.db to /root/trilium-data/backup : KO !

**Reply by tteck** [reply]
Restore a backup. You do have backups, right?

**Reply by tteck** [reply]
https://github.com/TriliumNext/Notes/blob/develop/README.md#migrating-from-trilium 🤷🏻‍♂️

**Reply by johndoe0815** [reply]
I know, but the odd thing that no one reports any problems there, at least if you are on the latest version. 
With the LXC and this script there are already several reports where it did not work.

**Reply by tteck** [reply]
Were you on the latest Trilium version before jumping to Trilium Next?

https://github.com/orgs/TriliumNext/discussions/369#discussioncomment-10501751

**Reply by johndoe0815** [reply]
Yes, see above my comments two weeks ago. That is why I mention it, I have seen that issue.

**Reply by tteck** [reply]
I'm sorry that it didn't work out for you, but there is nothing that I know to do.

**Reply by oOStroudyOo** [reply]
Honestly just run the following, I didn't lose any data and migrated from Trilium to TrilumNext.
Create a backup if you're worried.
```
systemctl stop trilium
rm -R /opt/trilium/
mkdir /opt/trilium/
bash -c "$(wget -qLO - https://github.com/tteck/Proxmox/raw/main/ct/trilium.sh)"
```

**Reply by johndoe0815** [reply]
I can confirm that this did work, thanks!
In you initial post I thought that this would loose the data and lead to a fresh trilium install, but indeed data is still there afterwards.

**Reply by dflahaut** [reply]
@oOStroudyOo : Many thanks ! It did work .
